src/hip_module.cpp

```diff
@@ -143,135 +143,116 @@ hipError_t ihipModuleLaunchKernel(TlsData *tls, hipFunction_t f, uint32_t global
     using namespace hip_impl;
 
     auto ctx = ihipGetTlsDefaultCtx();
-    hipError_t ret = hipSuccess;
 
-    if (ctx == nullptr) {
-        ret = hipErrorInvalidDevice;
+    if (!ctx) return hipErrorInvalidDevice;
 
-    } else {
-        int deviceId = ctx->getDevice()->_deviceId;
-        ihipDevice_t* currentDevice = ihipGetDevice(deviceId);
-        hsa_agent_t gpuAgent = (hsa_agent_t)currentDevice->_hsaAgent;
+    int deviceId = ctx->getDevice()->_deviceId;
+    ihipDevice_t* currentDevice = ihipGetDevice(deviceId);
+    hsa_agent_t gpuAgent = (hsa_agent_t)currentDevice->_hsaAgent;
 
-        std::vector<char> kernargs{};
-        if (kernelParams) {
-            if (extra) return hipErrorInvalidValue;
+    std::vector<char> kernargs{};
+    if (kernelParams && *kernelParams) {
+        if (extra) return hipErrorInvalidValue;
 
-            for (auto&& x : f->_kernarg_layout) {
-                const auto p{static_cast<const char*>(*kernelParams)};
+        for (auto&& x : f->_kernarg_layout) {
+            const auto p{static_cast<const char*>(*kernelParams)};
```

Comment:
@AlexVlx I think it will be good to check *kernelParams here. For example passing void* args[2] = {&Ad, nullptr}; will miss check at line 154 and will lead to seg fault.

Reply:
I don't think so. On one hand, the body of that branch is still broken anyway, so anybody using it would have a different set of problems. On the other, passing nullptr to this looks like UB, so you can get anything from nasal demons, segfaults or nuclear armageddon. We're choosing the option that makes the common case fast. You can, of course, follow-up and insert per element checks, just note that those would come at a cost.

Reply:
Ok but I tried that on CUDA that case and it throws an error however we seg faults.

Reply:
Yes...

Reply:
My basic point is that if needs be the check can be added at a later date, and since this is UB segfaulting is a perfectly kosher thing to do, orthogonally to what CUDA does or does not do. Indeed, adding the check would be a most excellent subsequent contribution.